### PR TITLE
Feature/tcskill

### PR DESCRIPTION
**Pull request by OliviaOstwald-NOAA**
## Pull Request Testing ##

- [ ] Describe testing already performed for this Pull Request:</br>
This PR is for generating stats and plots for TC spread/skill score metrics for global ensemble models. It has been tested using data from the 2023 Atlantic, East Pacific, and West Pacific tropical cyclones. 

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
Recommended testing is for the 2023 Hurricane Season, and that data can be found here: /lfs/h2/emc/vpppg/noscrub/olivia.ostwald/evs_tc_2023

To run this code package, clone input data directory and structure in personal noscrub space so no changes are necessary in the scripts. 

1. cd /lfs/h2/emc/vpppg/noscrub/$USER
2. mkdir evs_tc_2023
3. cp -r /lfs/h2/emc/vpppg/noscrub/olivia.ostwald/evs_tc_2023 /lfs/h2/emc/vpppg/noscrub/$USER/evs_tc_2023
4. Clone repository and run **jevs_hurricane_global_ens_spread_stats.sh** (EVS/dev/drivers/scripts/stats/hurricane). Once stats job has successfully finished, run plots job **jevs_hurricane_global_ens_spread_plots.sh**

- [x] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**

- [x] Do these updates/additions include sufficient testing updates? **[Yes or No]**

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

### CODE REVIEW

**Comment by malloryprow** [comment]
Hi Olivia! I'm looking over your changes. I see you added a directory "tc_verify" directory. This doesn't follow EE2's organization of the code repository. What is the purpose of this "tc_verify" directory? Was this something discussed previously of where to put this directory?

**Comment by OliviaOstwald-NOAA** [comment]
Thanks for pointing that out, it was a mistake! I had accidentally copied tc_verify in my EVS directory and forgot to remove it, but just did.

**Comment by malloryprow** [comment]
It also looks like there were regional stats and plots drivers deleted:

1. dev/drivers/scripts/plots/hurricane/jevs_hurricane_regional_late_tropcyc_ops_plots.sh
2. dev/drivers/scripts/plots/hurricane/jevs_hurricane_regional_early_tropcyc_legacy_plots.sh
3. dev/drivers/scripts/plots/hurricane/jevs_hurricane_regional_early_tropcyc_ops_plots.sh
4. dev/drivers/scripts/plots/hurricane/jevs_hurricane_regional_late_tropcyc_legacy_plots.sh
5. dev/drivers/scripts/stats/hurricane/jevs_hurricane_regional_late_tropcyc_ops_stats.sh
6. dev/drivers/scripts/stats/hurricane/jevs_hurricane_regional_early_tropcyc_ops_stats.sh
7. dev/drivers/scripts/stats/hurricane/jevs_hurricane_regional_early_tropcyc_ops_stats.sh
8. dev/drivers/scripts/stats/hurricane/jevs_hurricane_regional_late_tropcyc_legacy_stats.sh

Just asking since it looks like this PR is about adding spread and skill for global_ens models, but these unrelated regional scripts are being deleted.

**Comment by OliviaOstwald-NOAA** [comment]
That was intentional, here is the background on that: 
Last year (~Spring 2023), Logan and Geoff asked me to split the regional jobs into early and late model cycles for the "operational" and "legacy" models separately, resulting in four different drivers (and exevs scripts). The "regional_ops" package verified HAFS-A, HAFS-B, GFS, and CO-AMPS, while the "regional_legacy" package verified HAFS-A, HAFS-B, HWRF, and HMON. That made it into my original PR for the 2023 summary stats and plots which was submitted right before Geoff and Logan left and the original EVS code delivery date last September. 

After some time, I decided to combine "ops" and "legacy" since they both verified two of the same models (HAFS), and got word that HWRF and HMON would continue to run operationally for the 2024 season. This is [once again] said to be the last season with HWRF and HMON running, and can be updated next season if both models are turned off at the end of this season. 

So, I have been removing those scripts from newer feature branches (feature/2024tc for in-season analysis, feature/tcskill for global_ens) to keep things more organized for myself. I don't think that will impact this PR since the changes are in the global_ens_spread scripts which are all new codes, but I have very few PRs under my belt so please feel free to let me know if I made a mistake, or am going down the wrong path :)

**Comment by malloryprow** [comment]
Thanks! Nope that all sounds good. I'm good with including getting rid of those in PR, and I see how the changes in the J-Jobs follows that as well.

I looked over the code the rest of the code and it looks good to me, but `dev/modulefiles/hurricane/hurricane_stats.sh`. I left two comments there for correction.

**Comment by malloryprow** [comment]
Following up on my last comment...It looks like prod_envir is being loaded in jevs_hurricane_global_ens_spread_plots.sh and jevs_hurricane_global_ens_spread_stats.sh. It doesn't need to me loaded again. I'd remove this. Additionally, `module load prod_util/${produtil_ver}` should be `module load prod_util/${prod_util_ver}`.

File: dev/modulefiles/hurricane/hurricane_stats.sh
Comment (malloryprow) [inline dev/modulefiles/hurricane/hurricane_stats.sh:1]: Testing

**Comment by OliviaOstwald-NOAA** [comment]
Made those changes!

File: dev/drivers/scripts/plots/hurricane/jevs_hurricane_global_ens_spread_plots.sh
Comment (AliciaBentley-NOAA) [inline dev/drivers/scripts/plots/hurricane/jevs_hurricane_global_ens_spread_plots.sh]: @malloryprow @OliviaOstwald-NOAA Honest question for you. Should FIXevs point to the dev location of the fix directory or the ops EVS v1.0 location of the fix directory?

File: scripts/stats/hurricane/exevs_hurricane_global_ens_spread_stats.sh
Comment (AliciaBentley-NOAA) [inline scripts/stats/hurricane/exevs_hurricane_global_ens_spread_stats.sh]: @OliviaOstwald-NOAA Would it be possible for this line to use variables again, as it used to? Like ${STEP}/${COMPONENT} instead of stats/hurricane? 

CC @malloryprow

Comment (OliviaOstwald-NOAA) [reply scripts/stats/hurricane/exevs_hurricane_global_ens_spread_stats.sh]: Yep! I will look for other lines where I may have missed any hardcoded variables too, but feel free to point any out that you come across before I do.

File: dev/drivers/scripts/plots/hurricane/jevs_hurricane_global_ens_spread_plots.sh
Comment (malloryprow) [reply dev/drivers/scripts/plots/hurricane/jevs_hurricane_global_ens_spread_plots.sh]: I had thought about it. Technically it doesn't need to be defined since FIXevs gets defined in the J-Job as $HOMEevs/fix, and that is usually just a soft link to /lfs/h2/emc/vpppg/noscrub/emc.vpppg/verification/EVS_fix. Here is just pointing directly to it. If this ran in ops, I would say it should be removed but since this doesn't I'm fine with keeping it or not.

**Comment by OliviaOstwald-NOAA** [comment]
@AliciaBentley-NOAA just pushed that revision, I didn't find any other lines that I missed but please let me know if you come across any and I will update them as well. Thanks for catching that!

**Comment by AliciaBentley-NOAA** [comment]
Thanks, @OliviaOstwald-NOAA! Those two comments (FIXevs and adding ${STEP}/${COMPONENT}) were the only two things I saw on my cursory review! I don't think I'm assigned to this PR, so that's likely all from me! Thanks! 

CC @malloryprow

**Comment by malloryprow** [comment]
Everything looks good then! We are going to put testing on hold until @OliviaOstwald-NOAA returns from vacation. The stats job takes a few hours to run and I'm heading out over the weekend at 3:30. But with everything looking good we will be ready to dive right in when she gets back!

**Comment by OliviaOstwald-NOAA** [comment]
Sounds good! I don't know enough about FIXevs, or the dev vs ops directories so I will defer to Mallory (or Shelley or Perry) for that. Have a great weekend and we'll be in touch when I'm back!

**Comment by malloryprow** [comment]
@OliviaOstwald-NOAA I'm going to take another look over the code, but everything good with you if we start testing this PR today?

File: dev/drivers/scripts/plots/hurricane/jevs_hurricane_global_ens_spread_plots.sh
Comment (malloryprow) [reply dev/drivers/scripts/plots/hurricane/jevs_hurricane_global_ens_spread_plots.sh]: Let's go ahead and get rid of `export FIXevs=/lfs/h2/emc/vpppg/noscrub/emc.vpppg/verification/EVS_fix`

**Comment by malloryprow** [comment]
And please sync your fork's develop branch and then merge develop into this branch!

File: ush/hurricane/plot_tropcyc_lead_spd.py
Comment (malloryprow) [inline ush/hurricane/plot_tropcyc_lead_spd.py:1]: I know this doesn't run in ops, but let's keep it as consistent as we can for other operational standards. Put `#!/usr/bin/env python3` as the first line of the script.

File: ush/hurricane/plot_tropcyc_lead_spread.py
Comment (malloryprow) [inline ush/hurricane/plot_tropcyc_lead_spread.py:1]: I know this doesn't run in ops, but let's keep it as consistent as we can for other operational standards. Put `#!/usr/bin/env python3` as the first line of the script.

File: dev/drivers/scripts/plots/hurricane/jevs_hurricane_global_ens_spread_plots.sh
Comment (malloryprow) [reply dev/drivers/scripts/plots/hurricane/jevs_hurricane_global_ens_spread_plots.sh]: Can we remove `##PBS -l place=vscatter:exclhost,select=1:ncpus=128:ompthreads=1` to avoid confusion on resources being used?

**Comment by malloryprow** [comment]
Sorry @OliviaOstwald-NOAA may need to do the sync once more as another one just got merged!

**Comment by OliviaOstwald-NOAA** [comment]
> Sorry @OliviaOstwald-NOAA may need to do the sync once more as another one just got merged!

No problem at all!! I just logged in for the first time in about a week, so lots up updates to sync, thanks for that heads up!

File: ush/hurricane/plot_tropcyc_lead_spd.py
Comment (OliviaOstwald-NOAA) [reply ush/hurricane/plot_tropcyc_lead_spd.py:1]: So this script was in Jiayi's original repo and I had copied it into my fork before settling on using **plot_tropcyc_lead_spread.py**. Is it okay if I delete this script? And will add this line to plot_tropcyc_lead_spread.py

Comment (malloryprow) [reply ush/hurricane/plot_tropcyc_lead_spd.py:1]: Yes, if it isn't being used delete it please!

**Comment by malloryprow** [comment]
Noticing a few more things... I'll share them here since it seems GitHub is acting a little weird on the comments on the file directly

jevs_hurricane_global_ens_spread_plots.sh: remove line 13 (#%include <head.h>), remove line 14 (#%include <envir-p1.h>), remove line 56 (%include <tail.h>), remove line 57 (%manual), remove line 59 (%end)

jevs_hurricane_global_ens_spread_stats.sh: remove line 13 (#%include <head.h>), remove line 14 (#%include <envir-p1.h>), remove line 54 (%include <tail.h>), remove line 55 (%manual), remove line 57 (%end)

These are all things related to ecflow not need for the dev driver scripts.

**Comment by malloryprow** [comment]
Thanks for the changes @OliviaOstwald-NOAA! I'm going to pull them in and gets the stats job set up and running.

**Comment by malloryprow** [comment]
Job is underway.

**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr507/EVS/dev/drivers/scripts/stats/hurricane/jevs_hurricane_global_ens_spread_stats.o147687219
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr507/evs/v2.0/stats/hurricane/global_ens/spread
**DATA**: /lfs/h2/emc/stmp/mallory.row/jevs_hurricane_global_ens_spread_stats_00.147687219.cbqs01

I'll note two things I noticed that are set different in the hurricane drivers in this PR than other components:

1.  `DATAROOT=/lfs/h2/emc/stmp/$USER` in the hurricane drivers, whereas in other components it is `DATAROOT=/lfs/h2/emc/stmp/${USER}/evs_test/$envir/tmp`.
2. `#PBS -A ENSTRACK-DEV` in the hurricane drivers, whereas in other components it is `#PBS -A VERF-DEV`.

If this is consistent across the hurricane drivers, these updates can be addressed in a separate PR so they can be addressed for all the hurricane drivers.

**Comment by malloryprow** [comment]
I'm heading out for the weekend soon, but it looks like the job is almost 2 hours and 30 minutes in. Not sure what the normal run time is.

**Comment by OliviaOstwald-NOAA** [comment]
That's a good sign! It has been taking ~4.5 hours to run the stats job, and I checked the logfile and COMOUT paths and see the METplus tc_stat and tc_pairs files populating in their respective directories. Once stats is done, the plots job takes under an hour (I believe no more than ~20 minutes) so it will be much faster to test. 

The completed output looks good and matches that from my runs. Have a great weekend!

File: dev/drivers/scripts/plots/hurricane/jevs_hurricane_global_ens_spread_plots.sh
Comment (malloryprow) [inline dev/drivers/scripts/plots/hurricane/jevs_hurricane_global_ens_spread_plots.sh:1]: `#PBS -l select=1:ncpus=2:mem=4GB` should be `#PBS -l select=1:ncpus=1:mem=4GB`. I don't see any parallelization being used (unless I missed it), so resources should be set for serial run.

File: dev/drivers/scripts/stats/hurricane/jevs_hurricane_global_ens_spread_stats.sh
Comment (malloryprow) [inline dev/drivers/scripts/stats/hurricane/jevs_hurricane_global_ens_spread_stats.sh:1]: `#PBS -l select=1:ncpus=2:mem=4GB` should be `#PBS -l select=1:ncpus=1:mem=4GB`. I don't see any parallelization being used (unless I missed it), so resources should be set for serial run.

**Comment by malloryprow** [comment]
I ran jevs_hurricane_global_ens_spread_plots.sh but it seemed to run pretty fast. Take a look and let me know.

**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr507/EVS/dev/drivers/scripts/plots/hurricane/jevs_hurricane_global_ens_spread_plots.o148091345
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr507/evs/v2.0/plots/hurricane/global_ens/spread
**DATA:** /lfs/h2/emc/stmp/mallory.row/jevs_hurricane_global_ens_spread_plots_00.148091345.cbqs01

I ran with 

**COMINstats:**/lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr507/evs/v2.0/stats/hurricane/global_ens/spread
**COMINvit:**/lfs/h2/emc/vpppg/noscrub/olivia.ostwald/evs_tc_2023/syndat_tcvitals.2023
**COMINbdeckNHC:**/lfs/h2/emc/vpppg/noscrub/olivia.ostwald/evs_tc_2023/bdeck
**COMINbdeckJTWC:**/lfs/h2/emc/vpppg/noscrub/olivia.ostwald/evs_tc_2023/bdeck

**Comment by OliviaOstwald-NOAA** [comment]
The plots look good **but** the colors are incorrect for MD04-MD08 (UENSerror, all three spread). I am correcting those settings as well as "#PBS -l select=1:ncpus=1:mem=4GB"

Also, the plots job does take much less time to run!

**Comment by malloryprow** [comment]
I re-ran jevs_hurricane_global_ens_spread_plots.sh with the new changes.

Log File: /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr507/EVS/dev/drivers/scripts/plots/hurricane/jevs_hurricane_global_ens_spread_plots.o148108549
COMOUT: /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr507/evs/v2.0/plots/hurricane/global_ens/spread
DATA: /lfs/h2/emc/stmp/mallory.row/jevs_hurricane_global_ens_spread_plots_00.148108549.cbqs01

**Comment by OliviaOstwald-NOAA** [comment]
Checked COMOUT and the plots look good!

**Comment by malloryprow** [comment]
Great! This PR is good for me then. I'm going to approve. @AliciaBentley-NOAA

**Review by malloryprow — APPROVED** [review]

**Review by AliciaBentley-NOAA — APPROVED** [review]
@malloryprow @OliviaOstwald-NOAA I have reviewed all of the changes made in this PR and found that they looked correct and made sense. I approve this PR to be merged into develop. Thanks!

**Pull request merged**
